vmm/src/vm.rs

```diff
@@ -1077,6 +1077,14 @@ impl Vm {
 
         let pci_space = (pci_space_start.0, pci_space_size);
 
+        let virtio_iommu_bdf = if let Some((virtio_iommu_bdf, _)) =
+            self.device_manager.lock().unwrap().iommu_attached_devices()
+        {
+            Some(*virtio_iommu_bdf)
+        } else {
+            None
+        };
```

Comment:
You can simplify with `map()` here:
```rust
let virtio_iommu_bdf = self.device_manager.lock().unwrap().iommu_attached_devices().map(|(v, _)| *v);
```

Reply:
Thank you for the graceful simplification.